debugtools/DDR_VM/src/com/ibm/j9ddr/tools/ddrinteractive/commands/PatchMemoryCommand.java

```diff
@@ -0,0 +1,193 @@
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package com.ibm.j9ddr.tools.ddrinteractive.commands;
+
+import java.io.PrintStream;
+
+import com.ibm.j9ddr.corereaders.memory.MemoryFault;
+import com.ibm.j9ddr.corereaders.memory.MemoryRecord;
+import com.ibm.j9ddr.corereaders.memory.PatchedMemoryList;
+import com.ibm.j9ddr.tools.ddrinteractive.Command;
+import com.ibm.j9ddr.tools.ddrinteractive.Context;
+import com.ibm.j9ddr.tools.ddrinteractive.DDRInteractiveCommandException;
+import com.ibm.j9ddr.tools.ddrinteractive.ExpressionEvaluator;
+import com.ibm.j9ddr.tools.ddrinteractive.ExpressionEvaluatorException;
+
+/**
+ * Commands to patch the memory.
+ * <p>
+ * @author Manqing LI, IBM Canada
+ *
+ */
+public class PatchMemoryCommand extends Command {
+
+	public static final String COMMAND_SET = "setMemory";
+	public static final String ARGUMENTS_SET = "<targetAddress> <byte>[ <byte> ...]";
+	public static final String ARGUMENTS_DESCRIPTION_SET = "Set the memory starting at <targetAdderss> to the specified bytes.";
+	
+	public static final String COMMAND_REMAP = "remapMemory";
+	public static final String COMMAND_REMAP_SHORT = "remap";
+	public static final String ARGUMENTS_REMAP = "<sourceAddress> <length> <targetAddress>";
+	public static final String ARGUMENTS_DESCRIPTION_REMAP = "Remap <length> bytes from <sourceAddress> to <targetAddress>.";
+
+	public static final String COMMAND_CLEAR = "clearPatchedMemory";
+	public static final String ARGUMENTS_CLEAR = "[<address>[,<length>] ...]";
+	public static final String ARGUMENTS_DESCRIPTION_CLEAR = "Clear all the patched memory, or only the patched memory for the specified address.";
+
+	public static final String COMMAND_LIST = "listPatchedMemory";
+	public static final String ARGUMENTS_LIST = "";
+	public static final String ARGUMENTS_DESCRIPTION_LIST = "Display all the current patched memory records.";
+
+	public static final String COMMAND_VERBOSE = "verbosePatchedMemory";
+	public static final String ARGUMENTS_VERBOSE = "[<No>|<False>]";
+	public static final String ARGUMENTS_DESCRIPTION_VERBOSE = "Print verbose messages for the patched memory, or do not print verbose messages if <No> or <False> is specified.";
+
+	public PatchMemoryCommand() {
+		addCommand(COMMAND_SET, ARGUMENTS_SET, ARGUMENTS_DESCRIPTION_SET);
+		addCommand(COMMAND_REMAP, ARGUMENTS_REMAP, ARGUMENTS_DESCRIPTION_REMAP);
+		addCommand(COMMAND_REMAP_SHORT, ARGUMENTS_REMAP, ARGUMENTS_DESCRIPTION_REMAP);
+		addCommand(COMMAND_CLEAR, ARGUMENTS_CLEAR, ARGUMENTS_DESCRIPTION_CLEAR);
+		addCommand(COMMAND_LIST, ARGUMENTS_LIST, ARGUMENTS_DESCRIPTION_LIST);
+		addCommand(COMMAND_VERBOSE, ARGUMENTS_VERBOSE, ARGUMENTS_VERBOSE);
```

Comment:
I suggest the set of commands should have a common prefix:
* patch list
* patch set ...
* patch clear ...
...